src/tasks/FileUploadUtil/UploadResult.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+/**
+ * Class representing a successful file upload result
+ */
+export class UploadResult {
+	/**
+	 * @private
+	 * Location value looked up in the response header
+	 */
+	private _location: string;
+
+	/**
+	 * @private
+	 * Response body of the final raw response
+	 */
+	private _responseBody: unknown;
+
+	/**
+	 * @public
+	 * Get of the location value.
+	 * Location value is looked up in the response header
+	 */
+	public get location(): string {
+		return this._location;
+	}
+
+	/**
+	 * @public
+	 * Set the location value
+	 * Location value is looked up in the response header
+	 */
+	public set location(location: string) {
+		this._location = location;
+	}
+
+	/**
+	 * @public
+	 * Get The response body from the completed upload response
+	 */
+	public get responseBody() {
+		return this._responseBody;
+	}
+
+	/**
+	 * @public
+	 * Set the response body from the completed upload response
+	 */
+	public set responseBody(responseBody: unknown) {
+		this._responseBody = responseBody;
+	}
+
+	/**
+	 * @public
+	 * @param {responsebody} responsebody - The response body from the completed upload response
+	 * @param {responsebody} location - The location value from the headers from the completed upload response
+	 */
+	public constructor(responseBody: unknown, location: string) {
+		// Response body or the location parameter can be undefined.
+		this._location = location;
+		this._responseBody = responseBody;
+	}
+
+	/**
+	 * @public
+	 * @param {responsebody} responsebody - The response body from the completed upload response
+	 * @param {responsebody} responseHeaders - The headers from the completed upload response
+	 */
+	public static CreateUploadResult(responsebody?: unknown, responseHeaders?: Headers) {
```

Comment:
any specific reason why this is not simply a constructor?

Reply:
Yes, the intention is to keep the implementation open say if there is more than one way or more than a single source to populate the UploadResult object.

Reply:
because of the single constructor limitation? What do you think of using a constructor with optional parameters and default values? This way the caller would use a standard way of creating the object, pass-in what they can. Or even better, using a local initialization interface like [described here](https://betterprogramming.pub/named-parameters-in-typescript-e32c763d2b2e) ? (this would effectively add a behavior like named parameters, be extensible, and it's a very common practice in TS)

Reply:
Yes right I did this because of the single constructor limitation. 

I prefer the static factory pattern here - 

I did consider named parameters or having an interface with default/optional parameters. My thought is that if tomorrow I have to add another way of populating the UploadResult adding the logic in a separate method would be clean and also locating the method will be easier. Something like `CreateUploadResultWithNewAPIResult()` , `CreateUploadResultWithBetaResult()`. The name of the factory method will clarify what we need and its easier to search for a method than go through list of parameters and set what is needed. Also, grouping related parameters for a named structure would be clean. I don't want to group unrelated parameters.

> Consider public static methods which return a new Matrix instead of constructors overloads.
https://codereview.stackexchange.com/questions/124849/emulating-constructor-overloading-in-typescript

Reply:
Yeah I guess I'm coming at "too much of a react angle" for the interface and objects deconstruct approach.
I can't see an obvious benefit for one or the other besides conventions and preferences :)
last thing, can you change responsebody to responseBody for consistency with the other parameter please?

Reply:
Changed to responseBody!